snuba/admin/clickhouse/migration_checks.py

```diff
@@ -0,0 +1,215 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Sequence, Tuple
+
+from snuba.admin.migrations_policies import get_migration_group_polices
+from snuba.migrations.groups import MigrationGroup
+from snuba.migrations.policies import MigrationPolicy
+from snuba.migrations.runner import MigrationDetails, MigrationKey
+from snuba.migrations.status import Status
+
+
+class Reason(Enum):
+    NO_REASON_NEEDED = ""
```

Comment:
`Reason.no_reason` seems confusing. There is a clear way to identify a case where there should be no reason, which is making the reason optional.